### DISCUSSION TITLE
Pre hook for console mode, similar to splash feature for UI

### DISCUSSION THREAD

**Discussion by kk-hiraskar** [pyinstaller/pyinstaller#6130] [Features & Development]
How to add a pre hook or some script which does basic things like just printing something on console at the extreme beginning step.

Because on console applications (in onefile mode) until extractions is done, a blank console can confuse user,
so want to add few print statements, similar concept of splash which serves for window mode.

**Comment by rokm** [comment]
There are no splash-screen-equivalent messaging facilities for console mode (unless you count debug messages that can be enabled via `--debug bootloader`...). All execution prior to start up of embedded python interpreter that runs your python script happens in the bootloader executable (which is the actual executable part of the frozen application).

So if you want custom messages printed during start-up phase (before your python script is executed), you will need to add corresponding `printf()` statements to bootloader code, recompile the bootloader, and then rebuild your application so that it uses your modified bootloader.

**Reply by kk-hiraskar** [reply]
Ok..

As we already have with GUI in form of splash, definitely that feature is a more bigger than this just print thing.

is it possible to make this as an option, can we propose this feature for pyinstaller ?

**Reply by bwoodsend** [reply]
The splash screen is able to be there before loading the Python interpreter because there's a monster pile of C code baked into the bootloader which is able to detect and show a splash screen image without depending on Python. If you wanted stuff printed, you'd need to do something similar. I suppose just a piece of text wouldn't be too hard but there's very little scope for *custom behaviour* without editing and recompiling the C code and it would have to not depend on Python.
